packages/teleterm/src/ui/DocumentTerminal/useDocTerminal.ts

```diff
@@ -47,24 +48,50 @@ export default function useDocumentTerminal(doc: Props['doc']) {
   }
 
   function updateDocumentPid(pty: PtyProcess): void {
-    ctx.serviceDocs.update(doc.uri, { pid: pty.getPid() });
+    const pid = pty.getPid();
+    if (pid) {
+      ctx.serviceDocs.update(doc.uri, { pid });
+    }
+  }
+
+  function openTerminalContextMenu(): void {
+    ctx.mainProcessClient.openTerminalContextMenu();
   }
 
+  const refreshDocumentCwd = useMemo(() => {
+    return debounce(async () => {
+      const pid = ptyProcess.getPid();
+      const cwd = await ctx.serviceTerminals.getWorkingDirectory(pid);
+      ctx.serviceDocs.update(doc.uri, { cwd, title: cwd });
+    }, 2000);
+  }, [ptyProcess]);
+
   useEffect(() => {
+    let disposed = false;
     if (!ptyProcess) {
       createPtyProcess().then(pty => {
-        setPtyProcess(pty);
-        updateDocumentPid(pty);
+        if (!disposed) {
+          setPtyProcess(pty);
+          updateDocumentPid(pty);
+        }
       });
     }
 
+    ptyProcess?.onExit(() => {
```

Comment:
lets check for the exit code `0` before closing the tab. 
if a process exits with an error code. it's better to keep the tab open to notify a user about it.

Reply:
Done